Integer overflow in `Cursor::new()`

### Describe the bug
There is a integer overflow in `Cursor::new()` at ostd/src/mm/page_table/cursor.rs:175 when Asterinas trying to access a low memory address which is set by syscall `mmap` with empty file.

https://github.com/asterinas/asterinas/blob/963874471284ed014b76d268d933b6d13073c2cc/ostd/src/mm/page_table/cursor.rs#L175

### To Reproduce
1. Compile a program which calls `mmap` with empty file and access low memory address: 
```C
#include <errno.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <sys/mman.h>
#include <unistd.h>

struct MyStructure {
  int myNum;
  char myLetter;
};

int main() {
  int fd = -1;
  struct MyStructure *p = NULL;

  FILE *f = fopen("./testfile-2", "w+");
  fd = fileno(f);
  void* addr = mmap(NULL, 0x1000, 0x3, 0x4b112, fd, 0x0);
  perror("mmap");
  printf("address: %p\n", addr);
  printf("%d\n", p->myNum);

  return 0;
}
```
2. Run the compiled program in Asterinas.


### Expected behavior

Asterinas reports subtract with overflow and is terminated.

### Environment
- Official docker asterinas/asterinas:0.8.0
- 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
- Asterinas version: main 9638744

### Logs
```
panicked at /root/asterinas/ostd/src/mm/page_table/cursor.rs:175:46:
attempt to subtract with overflow
Printing stack trace:
   1: fn 0xffffffff88826290 - pc 0xffffffff888262fa / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bee20;

   2: fn 0xffffffff888260a0 - pc 0xffffffff888261f0 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bee80;

   3: fn 0xffffffff88049000 - pc 0xffffffff8804900a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8befd0;

   4: fn 0xffffffff889c2c50 - pc 0xffffffff889c2cd2 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8befe0;

   5: fn 0xffffffff889a0260 - pc 0xffffffff889a028e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bf070;

   6: fn 0xffffffff8882e8c0 - pc 0xffffffff8882ebec / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bf0c0;

   7: fn 0xffffffff88830df0 - pc 0xffffffff88830e18 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bf310;

   8: fn 0xffffffff888aee40 - pc 0xffffffff888aee5a / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bf3a0;

   9: fn 0xffffffff888964a0 - pc 0xffffffff888964d2 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bf3c0;

  10: fn 0xffffffff8858ef60 - pc 0xffffffff8858f44c / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bf5a0;

  11: fn 0xffffffff8858e090 - pc 0xffffffff8858e24b / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bfa80;

  12: fn 0xffffffff88555c60 - pc 0xffffffff885560b4 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8bff30;

  13: fn 0xffffffff882efdb0 - pc 0xffffffff882efeb1 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c00b0;

  14: fn 0xffffffff8855ac20 - pc 0xffffffff8855aff6 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c01c0;

  15: fn 0xffffffff8855ab10 - pc 0xffffffff8855abec / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c0500;

  16: fn 0xffffffff885d7790 - pc 0xffffffff885d8326 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c0570;

  17: fn 0xffffffff8837c890 - pc 0xffffffff8837c89e / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c0f90;

  18: fn 0xffffffff887ed560 - pc 0xffffffff887ed576 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c0fb0;

  19: fn 0xffffffff887b4040 - pc 0xffffffff887b40a9 / registers:

     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c0fd0;


     rax                0x1; rdx                0x8; rcx                0x0; rbx                0x0;
     rsi                0x1; rdi 0xffffffff887ee247; rbp                0x0; rsp 0xffff80000b8c1000;

[OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panicking.rs:113
(  2) /root/asterinas/ostd/src/panicking.rs:59
(  3) 612kfi8k15yg732ys5rlrja7q:?
(  4) ??:?
(  5) ??:?
(  6) /root/asterinas/ostd/src/mm/page_table/cursor.rs:175
(  7) /root/asterinas/ostd/src/mm/page_table/cursor.rs:399
(  8) /root/asterinas/ostd/src/mm/page_table/mod.rs:244
(  9) /root/asterinas/ostd/src/mm/vm_space.rs:106
( 10) /root/asterinas/kernel/src/vm/vmar/vm_mapping.rs:320
( 11) /root/asterinas/kernel/src/vm/vmar/vm_mapping.rs:231
( 12) /root/asterinas/kernel/src/vm/vmar/mod.rs:327
( 13) /root/asterinas/kernel/src/vm/vmar/static_cap.rs:183
( 14) /root/asterinas/kernel/src/thread/exception.rs:58
( 15) /root/asterinas/kernel/src/thread/exception.rs:21
( 16) /root/asterinas/kernel/src/thread/task.rs:68
( 17) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:79
( 18) /root/.rustup/toolchains/nightly-2024-06-20-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2077
( 19) /root/asterinas/ostd/src/task/mod.rs:175
make: *** [Makefile:173: run] Error 1
```

--- a/ostd/src/mm/page_table/cursor.rs
+++ b/ostd/src/mm/page_table/cursor.rs
@@ -141,7 +141,7 @@ where
     /// Note that this function does not ensure exclusive access to the claimed
     /// virtual address range. The accesses using this cursor may block or fail.
     pub fn new(pt: &'a PageTable<M, E, C>, va: &Range<Vaddr>) -> Result<Self, PageTableError> {
-        if !M::covers(va) {
+        if !M::covers(va) || va.is_empty() {
             return Err(PageTableError::InvalidVaddrRange(va.start, va.end));
         }
         if va.start % C::BASE_PAGE_SIZE != 0 || va.end % C::BASE_PAGE_SIZE != 0 {

